doc/manual/parallel-computing.rst

```diff
@@ -455,21 +455,84 @@ points: in this case, when :func:`remotecall_fetch` is called.
 
 Channels
 --------
-Channels provide for a fast means of inter-task communication. A
-``Channel{T}(n::Int)`` is a shared queue of maximum length ``n``
-holding objects of type ``T``. Multiple readers can read off the :class:`Channel`
-via :func:`fetch` and :func:`take!`. Multiple writers can add to the :class:`Channel` via
-:func:`put!`. :func:`isready` tests for the presence of any object in
-the channel, while :func:`wait` waits for an object to become available.
-:func:`close` closes a :class:`Channel`. On a closed :class:`Channel`\, :func:`put!` will fail,
-while :func:`take!` and :func:`fetch` successfully return any existing values
-until it is emptied.
+The section on Tasks in :ref:`man-control-flow`, discussed the execution of
+multiple functions in a co-operative manner. :class:`Channels` can be quite useful
+to pass data between running tasks, particularly those involving I/O operations.
+
+Examples of operations involving I/O include reading/writing to files,
+accessing web services, executing external programs, etc. In all
+these cases, overall execution time can be improved if other tasks can be run
+while a file is being read, or we are waiting for an external service/program to complete.
+
+A channel can be visualized as a pipe, i.e., it has a write end and read end.
+- Multiple writers in different tasks can write to the same channel concurrently via
+  :func:`put!` calls.
+- Multiple readers in different tasks can read data concurrently via :func:`take!` calls.
+- Channels are created via the :class:`Channel{T}(sz)` constructor. The channel will only hold
+  objects of type ``T``. If the type is not specified, the channel can hold objects of
+  any type. ``sz`` refers to the maximum number of elements that can be held in the channel
+  at any time.
+- If a channel is empty, readers (on a :func:`take!` call) will block till data is available.
+- If a channel is full, writers (on a :func:`put!` call) will block till space becomes available.
+- :func:`isready` tests for the presence of any object in the channel, while :func:`wait`
+  waits for an object to become available.
+- :func:`close` closes a :class:`Channel`. On a closed :class:`Channel`\, :func:`put!` will fail,
+  while :func:`take!` and :func:`fetch` successfully return any existing values till
```

Comment:
This is very dense and could also really, really benefit from an example with comments. Also,

":func:`put` will fail on a closed `Channel`, while :func:`take!` and :func:`fetch` successfully return any existing values till the closed `Channel` is emptied"